sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -447,7 +447,11 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
       // Create the table if the table didn't exist.
       if (!tableExists) {
         val schema = JdbcUtils.schemaString(df, url)
-        val sql = s"CREATE TABLE $table ($schema)"
+        // To allow certain options to append when create a new table, which can be
+        // table_options or partition_options.
+        // E.g., "CREATE TABLE t (name string) ENGINE=InnoDB DEFAULT CHARSET=utf8"
+        val createtblOptions = extraOptions.getOrElse("createTableOptions", "")
```

Comment:
I am just wondering if it is possible to put the option in `JDBCOptions` (maybe with `truncate` as well). Also, I guess i'd be better if this option is documented in both `DataFrameWriter` and `readwriter.py`. For example `truncate` is documented above. 

In addition, I just found the documentation for `truncate` is missed in `readwriter.py`. It might be nicer if this one is fixed together here.